### DISCUSSION TITLE
Building with CUDA Host SDK

### DISCUSSION THREAD

**Discussion by nlbutts** [OE4T/meta-tegra#731] [General]
I'm trying to get the host SDK to include the CUDA tools. I already have nvcc and libraries installed on my Ubuntu 18.04 system, but I would like the tools and libraries be included in the SDK build. I've added 
`nativesdk-packagegroup-cuda-sdk-host`
to
`TOOLCHAIN_HOST_TASK_append`
But I get the following error

`WARNING: Unable to get checksum for cuda-binaries-ubuntu1804-native SRC_URI entry cuda-repo-ubuntu1804-10-2-local-10.2.89-440.40_1.0-1_amd64.deb: file could not be found
WARNING: cuda-binaries-ubuntu1804-native-10.2.89-1-r0 do_fetch: Failed to fetch URL file:///home/nlbutts/projects/maza/maza-yocto/build/downloads/nvidia/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.40_1.0-1_amd64.deb, attempting MIRRORS if available
ERROR: cuda-binaries-ubuntu1804-native-10.2.89-1-r0 do_fetch: Fetcher failure: Unable to find file file:///home/nlbutts/projects/maza/maza-yocto/build/downloads/nvidia/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.40_1.0-1_amd64.deb anywhere. The paths that were searched were:
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804-10.2.89-1/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/files/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804-10.2.89-1/appareo
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804/appareo
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/files/appareo
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804-10.2.89-1/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/files/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804-10.2.89-1/x86-64
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804/x86-64
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/files/x86-64
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804-10.2.89-1/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804/
    /home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/files/
    /home/nlbutts/projects/maza/maza-yocto/build/downloads
ERROR: cuda-binaries-ubuntu1804-native-10.2.89-1-r0 do_fetch: Fetcher failure for URL: 'file:///home/nlbutts/projects/maza/maza-yocto/build/downloads/nvidia/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.40_1.0-1_amd64.deb'. Unable to fetch URL from any source.
ERROR: Logfile of failure stored in: /home/nlbutts/projects/maza/maza-yocto/build/tmp/work-shared/x86_64/cuda-binaries-10.2.89-1-r0/temp/log.do_fetch.63161
ERROR: Task (virtual:native:/home/nlbutts/projects/maza/maza-yocto/poky/../meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804_10.2.89-1.bb:do_fetch) failed with exit code '1'
NOTE: Tasks Summary: Attempted 2775 tasks of which 2768 didn't need to be rerun and 1 failed.
`

Am I suppose to manually download `cuda-repo-ubuntu1804-10-2-local-10.2.89-440.40_1.0-1_amd64.deb`?

**Comment by madisongh** [comment]
It used to be that way, since the host-side tools were only available via the NVIDIA SDK Manager.  NVIDIA started making them available for download recently, though, and some changes went in around the end of April into `dunfell-l4t-r32.4.3` and later branches to take advantage of that change.

So if you're using a current branch, have you updated it lately?

**Reply by nlbutts** [reply]
We plan to start testing the latest updates this week. So we will hold off on the SDK changes until then. Thank you.
